pinakes/main/approval/models.py

```diff
@@ -184,9 +185,124 @@ class Meta:
             ),
         ]
 
+    def move_internal_sequence(self, delta):
+        """
+        Move current record up or down relative to other records.
+        delta is an integer indictating how many other records should this
+        record be moved over upward (negative) or downward (positive).
+        -math.inf moves the record to top while math.inf moves it to bottom.
+
+        Internal_sequence is a column used to sort the workflows by users.
+        The user only needs to be able to adjust the sequence of the workflow
+        but does not care the actual value of the internal_sequence. When a new
+        workflow is created it is placed to the end of the list with an
+        internal_sequence number much greater than the value from previous tail
+        of the list. When workflow_a is moved to between workflow_b and
+        workflow_c, its internal_sequence is simply adjusted to the average
+        number of b and c unless the average becomes a decimal, in which case
+        all the internal_sequence numbers will get rebalanced with a big
+        interval again.
+        """
+        if delta == 0:
+            return
+
+        if delta > 0:
+            self.internal_sequence = self._internal_sequence_plus(delta)
+        else:
+            self.internal_sequence = self._internal_sequence_minus(-delta)
+
+    def _internal_sequence_plus(self, delta):
+        if delta == math.inf:
```

Comment:
Usage `math.inf` \ `-math.inf` constants for indication of a first \ last positions is obcure.
For example you could use enum or constant parameter for a direction indication (e.g. `FIRST`, `LAST`, `FORWARD`, `BACKWARD`)

Reply:
I like this idea, I think it would be easier to digest when reading through. I'm still not sure if `math.inf` and `-math.inf` are the "correct" values for the constant since they aren't integers, but it's much easier to understand that you're trying to compare against FIRST (or TOP).